server/tests/backends/sql_translator_unit_tests/test_base_translator.py

```diff
@@ -604,6 +607,24 @@ def test_text(base_translator: BaseTranslator, default_step_kwargs: dict[str, An
 
     assert ctx.selectable.get_sql() == expected_query.get_sql()
 
+    # with quotes
+    text = "Hello World ' there ' i' am"
+
+    step = steps.TextStep(text=text, new_column=new_column_name)
+    ctx = base_translator.text(step=step, columns=selected_columns, **default_step_kwargs)
+
+    expected_query = Query.from_(previous_step).select(
+        *selected_columns, functions.Cast(ValueWrapper(text), "TEXT").as_(new_column_name)
+    )
+
+    # ' are replaced with \' and not '' anymore
+    assert (
+        expected_query.get_sql()
+        == 'SELECT "name","pseudonyme",CAST(\'Hello World \\\' there \\\' i\\\' am\' AS TEXT) "name" FROM "previous_with"'
```

Comment:
```suggestion
        == r'''SELECT "name","pseudonyme",CAST('Hello World \' there \' i\' am' AS TEXT) "name" FROM "previous_with"'''
```

However this should only be the case with GBQ

Reply:
removed !
will relay on the integration test, thanks 🙏🏽